internal/provider/resource_packer_channel.go

```diff
@@ -49,31 +48,33 @@ func resourcePackerChannel() *schema.Resource {
 				ValidateDiagFunc: validateStringNotEmpty,
 			},
 			// Optional inputs
-			"iteration_assignment": {
-				Description: "The iteration assignment information that will be used to assign a completed iteration to the channel.",
+			"iteration": {
+				Description: "The iteration assigned to the channel.",
 				Type:        schema.TypeList,
 				MaxItems:    1,
 				Optional:    true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"fingerprint": {
-							ConflictsWith: []string{"iteration_assignment.0.id", "iteration_assignment.0.incremental_version"},
-							Description:   "The fingerprint of the iteration to assign to the channel.",
-							Type:          schema.TypeString,
-							Optional:      true,
+							Description:  "The fingerprint of the iteration assigned to the channel.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ExactlyOneOf: []string{"iteration.0.id", "iteration.0.fingerprint", "iteration.0.incremental_version"},
```

Comment:
To clarify what's being enforced here, there can only be one of `id` or `fingerprint` or `incremental_version` set at a time?

Reply:
Yea, that’s correct. A practico we should only be allowed to use one of the attributes for iteration assignment. 

Allowing the ability to set multiple seems confusing and introduces and exposes a limitation in the upstream API where if multiple params are in the request the id take precedence. Which is the reason why I use HasChanges to only select the input that has changed before posting to the API.